src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```
@@ -180,11 +178,21 @@ The .NET Foundation licenses this file to you under the MIT license.
     <Error Condition="'$(_WhereLinker)' != '0' and '$(CppCompilerAndLinkerAlternative)' == '' and '$(_IsApplePlatform)' != 'true'"
       Text="Requested linker ('$(CppLinker)') not found in PATH." />
 
-    <Exec Command="&quot;$(CppLinker)&quot; -fuse-ld=lld -Wl,--version" Condition="'$(LinkerFlavor)' == 'lld'" IgnoreExitCode="true" StandardOutputImportance="Low" ConsoleToMSBuild="true">
+    <Exec Command="&quot;$(CppLinker)&quot; -fuse-ld=lld -Wl,--version" Condition="'$(LinkerFlavor)' == 'lld' or ('$(LinkerFlavor)' == '' and $(CppCompilerAndLinker.Contains('clang')))"
```

```suggestion
    <Exec Command="&quot;$(CppLinker)&quot; -fuse-ld=lld -Wl,--version" Condition="'$(LinkerFlavor)' == 'lld'" IgnoreExitCode="true" StandardOutputImportance="Low" ConsoleToMSBuild="true">
```

Revert this line: this is not correct. Before we were checking the --version regardless of LinkerFlavor emptiness. Which means FreeBSD and user-supplied -p:LinkkerFlavor was also considered.

---

You are right about this one. We do not need to revert it.

When `LinkerFlavor` is `lld`, we will always check the version. Which means `_LinkerVersion` can't be empty if `lld` exists and changes made in `src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets` are not needed, right?

---

Then it will result in error one way or the other. We shouldn't be checking for emptiness.